src/shared/dissect-image.c

```diff
@@ -413,9 +413,11 @@ int dissect_image(
                         if (r < 0)
                                 return r;
 
+                        m->verity = !sd_id128_is_null(verity_uuid);
```

Comment:
hmm, verity_uuid is set exactly when root_hash is non-NULL. I think it would be much nicer to test for that here directly:
`
```c
m->verity = !!root_hash;
```

After all for your case the verity UUID is not actually used. After all we only use it as method to find the right root fs with the right verity, and look for the uuid in the gpt part table. But if you have no gpt table then the uuid is not useful. So the root hash is actually hhat you really want to check.

Reply:
good point, fixed